git/objects/commit.py

```diff
@@ -315,6 +315,18 @@ def stats(self) -> Stats:
             text = self.repo.git.diff(self.parents[0].hexsha, self.hexsha, '--', numstat=True)
         return Stats._list_from_string(self.repo, text)
 
+    @property
+    def patch(self) -> str:
```

Comment:
It probably would be worth describing that this method can fail if the processed bytes aren't in the encoding that python expects (as it returns `str` and not `bytes`). Hence this method can throw an exception and from my experience it's more than likely that this happens to some.

This also means that at some point in the future and if there is demand one could provide a variant of this method that returns bytes instead.

Reply:
Thanks for this note. I will check to see if i can return bytes than string and leave the choice to users to convert them to string. Is that okay?

Reply:
That would be okay, yes. If it doesn't work with returning `bytes` right now then the possibility of decoding errors could be highlighted in the documentation specifically.